cmd/launcher/doctor.go

```diff
@@ -0,0 +1,625 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/logutil"
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/pkg/agent/flags"
+	"github.com/kolide/launcher/pkg/agent/knapsack"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/autoupdate"
+	"github.com/kolide/launcher/pkg/autoupdate/tuf"
+	"github.com/kolide/launcher/pkg/launcher"
+	"github.com/kolide/launcher/pkg/log/checkpoint"
+	"github.com/peterbourgon/ff/v3"
+	"github.com/shirou/gopsutil/v3/process"
+
+	"github.com/fatih/color"
+	"golang.org/x/exp/slices"
+)
+
+var (
+	// Command line colors
+	cyanText   = color.New(color.FgCyan)
+	headerText = color.New(color.FgHiWhite).Add(color.Bold)
+	yellowText = color.New(color.FgHiYellow)
+
+	// Println functions for checkup details
+	whiteText = color.New(color.FgWhite)
+	greenText = color.New(color.FgGreen).PrintlnFunc()
+	redText   = color.New(color.FgRed).Add(color.Bold).PrintlnFunc()
+
+	// Indented output for checkup results
+	info = func(a ...interface{}) {
+		whiteText.Println(fmt.Sprintf("    %s", a...))
+	}
+	warn = func(a ...interface{}) {
+		yellowText.Println(fmt.Sprintf("    %s", a...))
+	}
+	fail = func(a ...interface{}) {
+		whiteText.Println(fmt.Sprintf("❌  %s", a...))
+	}
+	pass = func(a ...interface{}) {
+		whiteText.Println(fmt.Sprintf("✅  %s", a...))
+	}
+
+	configFile string
+	binDir     string
+)
+
+// checkup encapsulates a launcher health checkup
+type checkup struct {
+	name  string
+	check func() (string, error)
+}
+
+func runDoctor(args []string) error {
+	logger := log.With(logutil.NewCLILogger(true), "caller", log.DefaultCaller)
+	opts, err := parseDoctorOptions(os.Args[2:])
+	if err != nil {
+		level.Info(logger).Log("err", err)
+		os.Exit(1)
+	}
+
+	fcOpts := []flags.Option{flags.WithCmdLineOpts(opts)}
+	flagController := flags.NewFlagController(logger, nil, fcOpts...)
+	k := knapsack.New(nil, flagController, nil)
+
+	cyanText.Println("Kolide launcher doctor version:")
+	version.PrintFull()
+	cyanText.Println("\nRunning Kolide launcher checkups...")
+
+	checkups := []*checkup{
+		{
+			name: "Check platform",
+			check: func() (string, error) {
+				return checkupPlatform(runtime.GOOS)
+			},
+		},
+		{
+			name: "Check architecture",
+			check: func() (string, error) {
+				return checkupArch(runtime.GOARCH)
+			},
+		},
+		{
+			name: "Root directory contents",
+			check: func() (string, error) {
+				return checkupRootDir(getFilepaths(k.RootDirectory(), "*"))
+			},
```

Comment:
I haven't run this, but thinking aloud I see a difference in `doctor` and `flare`.

For `doctor`, I think this would just want to check if the directory exists. Maybe permissions. But `flare` would want to display the contents. I suspect with reasonable clever callers and signatures, this is supportable by the same underlying checkup function